crates/llama_cpp/src/session/completion.rs

```diff
@@ -26,7 +26,7 @@ impl CompletionHandle {
     /// Blocks the current thread, resolving to the next completed token, or `None` if EOS is
     /// reached.
     pub fn next_token(&mut self) -> Option<Token> {
-        block_on(self.rx.recv())
+        tokio::task::block_in_place(|| self.rx.blocking_recv())
```

Comment:
Why use block in place here? This function should be called in synchronous contexts only anyways.

Reply:
`blocking_recv` panics when called in an async context, so `block_in_place` allows `next_token` to block in an async context.

If you would prefer this function to panic, I'll remove `block_in_place` and update the documentation, but I'm concerned that making this function panic will be inconsistent with `LlamaSession::advance_context` and similar.

Reply:
Ah, I see, that's fair.
At some point, sync functions with async counterparts really need to be properly defined, but for now it's fine I guess.